### ISSUE TITLE
getting "Promise" is not defined     promise/no-native even with es6

### ISSUE THREAD

**Issue by gocarlos** [xjamundx/eslint-plugin-promise#154]
### Description

I'm getting the warning promise not defined, even though env is set to es6

### Steps to Reproduce
```json
    "env": {
        "es6": true,
        "mocha": true,
        "node": true

    },
   "rules": {
        "promise/always-return": "warn",
        "promise/no-return-wrap": "warn",
        "promise/param-names": "warn",
        "promise/catch-or-return": "warn",
        "promise/no-native": "warn",
        "promise/no-nesting": "warn",
        "promise/no-promise-in-callback": "warn",
        "promise/no-callback-in-promise": "warn",
        "promise/avoid-new": "warn",
        "promise/no-new-statics": "warn",
        "promise/no-return-in-finally": "warn",
        "promise/valid-params": "warn"
    },

```
**Expected behavior:**  
no warnings regarding promise not defined

**Actual behavior:**  
we get warnings...
### Versions

4.0.1 - latest at the time of writing...

- Node version: v10.15.0
- ESLint version:  5.12.0
- eslint-plugin-promise version: 4.0.1

**Comment by gocarlos** [comment]
I had that already in the configuration... 
Just realized that this warning is on porpose even on es6, thought that this warning would only apprear on es5... 


Still i do not get the reason for not using the native promise and instead use yet another package...

Nonetheless this is not related to this issue, thanks  @macklinu anyway for the quick response
